lua/roslyn/sln/utils.lua

```diff
@@ -41,27 +41,28 @@ function M.root_dir(buffer, broad_search)
         return name:match("%.csproj$") ~= nil
     end)
 
-    if not sln or not csproj then
+    if not sln and not csproj then
         return {}
     end
 
-    local projects = csproj and { files = find_files_with_extension(csproj, ".csproj"), directory = csproj } or nil
-
-    if broad_search then
-        local solutions = vim.fs.find(function(name, _)
-            return name:match("%.sln$")
-        end, { type = "file", limit = math.huge, path = sln })
-
-        return {
-            solutions = solutions,
-            projects = projects,
-        }
-    else
-        return {
-            solutions = find_files_with_extension(sln, ".sln"),
-            projects = projects,
-        }
+    local projects = nil
+    local solutions = nil
+    if csproj then
```

Comment:
Could you do a ternary here like the original code, and then do an early return if `sln` is nil?

Then you can also keep the if statement about broad_search exactly the same and avoid mutation of variables

Reply:
Makes sense, it makes the diff smaller. Done and rebased.